shell/platform/fuchsia/flutter/flatland_connection.cc

```diff
@@ -4,25 +4,20 @@
 
 #include "flatland_connection.h"
 
-#include <lib/fdio/directory.h>
-
 #include "flutter/fml/logging.h"
 
 namespace flutter_runner {
 
 FlatlandConnection::FlatlandConnection(
     std::string debug_label,
+    fidl::InterfaceHandle<fuchsia::ui::composition::Flatland> flatland,
     fml::closure error_callback,
     on_frame_presented_event on_frame_presented_callback,
     uint64_t max_frames_in_flight,
     fml::TimeDelta vsync_offset)
-    : error_callback_(error_callback),
+    : flatland_(flatland.Bind()),
+      error_callback_(error_callback),
       on_frame_presented_callback_(std::move(on_frame_presented_callback)) {
-  zx_status_t status =
-      fdio_service_connect("/svc/fuchsia.ui.composition.Flatland",
-                           flatland_.NewRequest().TakeChannel().release());
-  FML_DCHECK(status == ZX_OK);
-
   flatland_->SetDebugName(debug_label);
```

Comment:
Do you want to do flatland.set_error_handler() before starting to make calls on the channel?  If FIDL disconnects (e.x. scenic crashes) we aren't handling that case.

Reply:
Done.